include/vapor/SliceParams.h

```diff
@@ -40,18 +39,19 @@ class PARAMS_API SliceParams : public RenderParams {
             return GetVariableName();
     }
 
-    int GetSampleRate() const;
-
-    void SetSampleRate(int rate);
-
-    int GetDefaultSampleRate() const;
-
-    void SetCachedValues(std::vector<double> values);
-
+    int                 GetSampleRate() const;
+    void                SetSampleRate(int rate);
+    int                 GetDefaultSampleRate() const;
+    void                SetCachedValues(std::vector<double> values);
     std::vector<double> GetCachedValues() const;
 
     static const string _sampleRateTag;
-    static const string SampleLocationTag;
+    static const string XRotationTag;
```

Comment:
Simply moving these to RenderParams would make them usable by other renderers (e.g. contour renderer)

Reply:
I agree that the Contour and Slice renderers should have these tags.  But putting these tags into RenderParams makes it monolithic.  Most renderers have no dependency on rotation, so why should they keep in their state?

IE - The RenderParams base class should only have tags that are used among all of its subclasses.  Yet it has tags like _xFieldVariableNameTag. 
 Most renderers do not use field variables.  Unless I'm missing something, I believe this should be avoided.

Reply:
It's pretty common in OOP to have derived classes that don't use all of their superclasses capabilities. If these flags only applied to one renderer it would make sense to put them there. But since we've identified at least two in seems reasonable to make them common to the base class and have a getter that indicates if they are applicable (as we've done with field variables, height variables, etc.). The alternatives are 1) replication. Not good. 2) refactoring and introducing a new intermediate derived class every time a situation like this arises. Not practical

Reply:
Interesting.  Fixed.  Note that the SampleRateTag has also been added to RenderParams for the same reasons.

This kind of problem seems like a balancing act.  Borrowing from Stas' vehicle metaphor: 

Say we have a class Vehicle, with subclasses Car, Boat, and Airplane.  Boat and Airplane need a Rudder member.  The approach we're taking right now is to put Rudder in the Vehicle base class, even though Car doesn't use it.  This is fine if members like Rudder are reasonably limited, like in our case with Rotation and Origin tags.  However new intermediate classes eventually become necessary if Airplane needed members for ailerons, flaps, and the numerous components that don't exist on Cars and Boats.

I don't think we're at the point of adding things like "Ailerons and Flaps" to RenderParams yet.